main.go

```diff
@@ -46,5 +53,14 @@ func main() {
 	// Register template functions.
 	beego.AddFuncMap("i18n", i18n.Tr)
 
+	// serve securely if the certificates are present
+	_, certErr := os.Stat(servingCertFile)
+	_, keyErr := os.Stat(servingKeyFile)
+	if certErr == nil && keyErr == nil && len(servingCertFile) > 0 && len(servingKeyFile) > 0 {
```

Comment:
Hmm, maybe I should not be too focused on detail here, as I don't know exactly what you're going to use this for.
I noticed the current implementation would fail if any of those paths is a directory.

I'll drop a suggestion, @deads2k please let me know what you think:

```go
var (
	servingCertFile = os.Getenv("SERVING_CERT")
	servingKeyFile  = os.Getenv("SERVING_KEY")
)

func enableTLS() bool {
	for _, f := range []string{servingCertFile, servingKeyFile} {
		fi, err := os.Stat(f)
		if err != nil {
			return false
		}
		if !fi.Mode().IsRegular() {
			return false
		}
	}
	return true
}

func main() {
	// ...

	// serve securely if the certificates are present
	if enableTLS() {
		beego.HttpCertFile = servingCertFile
		beego.HttpKeyFile = servingKeyFile
		beego.EnableHttpTLS = true
	}

	beego.Run()
}
```

The code above would not accept symlinks (desirable?), only regular readable files.

@deads2k I'm ok with merging it as-is as well, just waiting for your reply :)